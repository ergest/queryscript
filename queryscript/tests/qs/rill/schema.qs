type Actor {
    aid text,
    url text,
    type text,
    gender text,
};

type Appearance {
    aid text,
    tid bigint,
    capacity text,
    role text,
    charid double,
    impid double,
    voice bool,
    epid bigint,
    sid bigint,
};

type Character {
    charid bigint,
    aid text,
    name text,
}

type Episode {
    sid bigint,
    epid bigint,
    aired text,
    epno bigint,
}

type Impression {
    impid bigint,
    aid text,
    name text,
}

export let actors [Actor] = load('data/actors.csv');
export let appearances [Appearance] = load('data/appearances.csv');
export let characters [Character] = load('data/characters.csv');
export let episodes [Episode] = load('data/episodes.csv');
export let impressions [Impression] = load('data/impressions.csv');
