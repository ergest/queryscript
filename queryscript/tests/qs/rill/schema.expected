{
    "compile_errors": [],
    "decls": {
        "let actors": [{
        	aid Utf8,
        	url Utf8,
        	type Utf8,
        	gender Utf8,
        }],
        "let appearances": [{
        	aid Utf8,
        	tid Int64,
        	capacity Utf8,
        	role Utf8,
        	charid Float64,
        	impid Float64,
        	voice Boolean,
        	epid Int64,
        	sid Int64,
        }],
        "let characters": [{
        	charid Int64,
        	aid Utf8,
        	name Utf8,
        }],
        "let episodes": [{
        	sid Int64,
        	epid Int64,
        	aired Utf8,
        	epno Int64,
        }],
        "let impressions": [{
        	impid Int64,
        	aid Utf8,
        	name Utf8,
        }],
        "type Actor": {
        	aid Utf8,
        	url Utf8,
        	type Utf8,
        	gender Utf8,
        },
        "type Appearance": {
        	aid Utf8,
        	tid Int64,
        	capacity Utf8,
        	role Utf8,
        	charid Float64,
        	impid Float64,
        	voice Boolean,
        	epid Int64,
        	sid Int64,
        },
        "type Character": {
        	charid Int64,
        	aid Utf8,
        	name Utf8,
        },
        "type Episode": {
        	sid Int64,
        	epid Int64,
        	aired Utf8,
        	epno Int64,
        },
        "type Impression": {
        	impid Int64,
        	aid Utf8,
        	name Utf8,
        },
    },
    "queries": [],
}